CONTRIBUTING.md

```diff
@@ -0,0 +1,249 @@
+Any fixtures, helper functions, constants or mocks usable by multiple tests should be added to the [utils](./testing/utils) subfolder.
+
+The folder structure should be:
+
+- `fixtures` for fixed data or examples
+- `utils` for utility functions
+
+The exception to this may be a utility which is applicable only to the current package.
+In this case, each package shall use the `test` folder to mirror the hierarchy of the `utils`
+TypeScript is set up to include this folder only for its test configuration.
+
+## Git
+
+Commit messages must follow [conventional commits](https://conventionalcommits.org/) format.
+In order to be prompted with supported types and scopes, stage your changes and run `npm run commit`.
+
+Branching strategy follows [trunk-based development](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development) guidelines.
+Pushing to remote triggers a CI workflow, which runs automated checks on your changes.
+
+The main branch should always have a linear history.
+Therefore, PRs are merged via one of two strategies:
+
+- rebase - branch cannot contain merge commits ([rebase instead of merge](https://www.atlassian.com/git/tutorials/merging-vs-rebasing)),
+- squash - single commit whose message is the PR title (should be in conventional commit format).
+
+## Project tags
+
+[Nx tags](https://nx.dev/core-features/enforce-module-boundaries) are used to enforce module boundaries in the project graph when linting.
+
+Projects are tagged in two different dimensions - scope and type:
+
+| tag                 | description                                                            | allowed dependencies               |
+| :------------------ | :--------------------------------------------------------------------- | :--------------------------------- |
+| `scope:shared`      | data models, utility functions, etc. (not specific to core or plugins) | `scope:shared`                     |
+| `scope:tooling`     | supplementary tooling, e.g. code generation                            | `scope:tooling`, `scope:shared`    |
+| `scope:internal`    | internal project, e.g. example plugin                                  | any                                |
+| `type:app`          | application, e.g. CLI or example web app                               | `type:util` or `type:testing-util` |
+| `type:util`         | general purpose utilities and types intended for reuse                 | `type:util` or `type:testing-util` |
+| `type:e2e`          | E2E testing                                                            | `type:app` or `type:testing-util`  |
+| `type:testing-util` | testing utilities                                                      | `type:util`                        |
+
+## Special targets
+
+The repository includes a couple of common optional targets:
+
+- `bench` - runs micro benchmarks of a project e.g. `nx bench utils` or `nx affected -t bench`
+
+## Special folders
```

Comment:
I’m not sure if we have all of these

Reply:
I will remove the so far unused folders. It was more about guidance for adding them. But let's cross the bridge when we are there